contracts/ArchetypeBatch.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+// ArchetypeBatch v0.1.0
+//
+//        d8888                 888               888
+//       d88888                 888               888
+//      d88P888                 888               888
+//     d88P 888 888d888 .d8888b 88888b.   .d88b.  888888 888  888 88888b.   .d88b.
+//    d88P  888 888P"  d88P"    888 "88b d8P  Y8b 888    888  888 888 "88b d8P  Y8b
+//   d88P   888 888    888      888  888 88888888 888    888  888 888  888 88888888
+//  d8888888888 888    Y88b.    888  888 Y8b.     Y88b.  Y88b 888 888 d88P Y8b.
+// d88P     888 888     "Y8888P 888  888  "Y8888   "Y888  "Y88888 88888P"   "Y8888
+//                                                            888 888
+//                                                       Y8b d88P 888
+//    
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+
+contract ArchetypeBatch is Ownable {
+
+    // Execute multiple calls in a single transaction
+    // targets: The addresses of the contracts
+    // values: The value to send to the contracts
+    // datas The data to send to the contracts
+    function executeBatch(
+        address[] calldata targets,
+        uint256[] calldata values,
+        bytes[] calldata datas
+    ) external payable {
+        require(
+            targets.length == values.length && targets.length == datas.length,
+            "ArchetypeBatch: The array lengths must be identical"
+        );
+
+        for (uint256 i = 0; i < targets.length; ++i) {
+            (bool success, bytes memory returnData) = targets[i].call{value: values[i]}(datas[i]);
+
+            if(!success) {
+                // Decode the return data as a string and revert with it
+                revert(string(returnData));
+            }
+        }
+    }
+
+    // Emergency function: In case any ETH get stuck in the contract unintentionally
+    // Only owner can retrieve the asset balance to a recipient address
+    function rescueETH(address recipient) onlyOwner external {
+        payable(recipient).call{ value: address(this).balance }("");
+    }
+
+    // Emergency function: In case any ERC20 tokens get stuck in the contract unintentionally
+    // Only owner can retrieve the asset balance to a recipient address
+    function rescueERC20(address asset, address recipient) onlyOwner external { 
```

Comment:
this will rescue coins from this batch contract only. Under what scenario would ETH/tokens get stuck here, if someone sends something here by accident?

Reply:
Yes if someone sends something by accident, or sends more eth then is defined in the values array they pass.